### PR TITLE
IBM Cloudant backed Sinopia 

### PR DESCRIPTION
**Pull request by shimmerjs**
Figured this would be a good as place as any for discussing what we need to do to our fork and try to align with your goals for the project overall.

### CODE REVIEW

**Comment by trentearl** [comment]
My goals are to have a private npm repository that can be be used by small companies and organizations. I dont have strong opinions beside that.

I would also like to have other maintainers of this project, so I would certainly love to have some volunteers.


@aeweidne would you like be added as an official maintainer?

----


There hasn't been a npm release of verdaccio so I dont think merging commits in has to be done with a lot of diligence, as long as the tests pass. I haven't checked recently but I dont think that either sinopia or verdaccio work on node6 at all, so we will have to figure that out anyway before an official release.

**Comment by aeweidne** [comment]
I would definitely love to be a contributor if you'll have me :).  We have definitely been discussing at least a Node LTS rewrite, being on pre 1.0 is gonna make it harder in the long run.

**Comment by jamietre** [comment]
Does this PR replace the default filesystem based data storage mechanism or is it configurable?

I'm just catching up on this.. I'm one of the many people who forked Sinopia b/c it wasn't really maintained. I am planning work on the UI in the next couple months since we are about to add a lot of users. I am trying to figure out the best place to do that, which hopefully would not just be in my own fork! But we don't need/want any change to the backend, the filesystem cache works fine for us and we like it because it's transparent and easy to maintain. Actually it didn't exactly work fine, because some people want to run it on Windows and it didn't really work on Windows before... fixing that was one of the reasons for my fork in the first place ;)

So basically just dropping in to try to figure out the direction and status of this in anticipation of some more work on it! Thanks for takling the lead on trying to revive Sinopa - it's an awesome project and it's always been frustrating that it was difficult to get anything moving in the main repo.

**Comment by andrewjensen** [comment]
I agree with @jamietre. I really like the filesystem-based storage because it's so easy to get up and running. If we add couchdb support, we should still keep FS storage as the default. Otherwise, we should probably maintain a `docker-compose.yml` to handle the burden of setting up couchdb.

**Comment by trentearl** [comment]
Yes I agree, I think including couchdb support could be good but file system should be default. I also dont want to obligate people to use docker, correct me if I wrong but wouldnt that mean verdaccio would only work on Linux?

**Comment by andrewjensen** [comment]
@trentearl currently, yes. Docker runs natively on Linux and their `docker-machine` integrations for Windows and Mac are really just an interface around a VirtualBox VM running Linux. Apparently Windows Server 2016 will [support docker](https://www.docker.com/microsoft) though, once it's out of Beta.

**Comment by aeweidne** [comment]
@jamietre In its current state this PR replaces the file system storage.  The intent is to make the storage system configurable before merging it in.

**Comment by jsumners** [comment]
+1 for keeping the file system storage. It's one of the biggest selling points for Sinopia.

**Comment by qballer** [comment]
+1 for file system. I don't mind helping out here and there. I forked sinopia for a bug fix and I really like it.

**Comment by juanpicado** [comment]
I wonder whether me implement a pluggable Storage as `auth` works since recent sinopia version how much interest would be to move this PR as an independent dependency.

**Comment by juanpicado** [comment]
I think it's completely possible plug a non-sql database as optional via plugins and let file system as default. See #103. I'll close this PR because is too old. I merged manually this PR with `origin/storage_pluggable`. Thanks @shimmerjs .
